Mock timeout scenario for default_location

christe6-osu opened the pull request:
General:

Added a test for the timeout scenario for default_location.
Modified default_location to include a try/except block to catch Timeout exceptions.

### Code:

1. [Y] Does your submission pass [tests](https://ryansurf.github.io/cli-surf/tests/)?
2. [Y] Have you run the [linter/formatter](https://ryansurf.github.io/cli-surf/styling/) on your code locally before submission?
3. [N/A] Have you updated the documentation/README to reflect your changes, as applicable?
4. [Y] Have you added an explanation of what your changes do?
5. [N/A] Have you written new tests for your changes, as applicable?

christe6-osu commented:
Testing for the Timeout exception required adding code to src/api.py, which I realize may not be desirable. It is passing the pre-existing tests and the new Timeout test. However, I have not been able to get the application to work locally, so I cannot run everything to verify functionality.

Please let me know how it looks and if there is anything you would like me to change.

ryansurf commented:
Hey @christe6-osu, I haven't had time to review this yet but wanted to let you know I've seen it! I'll get to it in the next few days

ryansurf commented:
> Testing for the Timeout exception required adding code to src/api.py, which I realize may not be desirable. It is passing the pre-existing tests and the new Timeout test. However, I have not been able to get the application to work locally, so I cannot run everything to verify functionality.
> 
> Please let me know how it looks and if there is anything you would like me to change.

Hey @christe6-osu, apologies for just getting to this. I'll go ahead and test this locally myself, but what is going wrong when you are trying to run it locally?

christe6-osu commented:
Sorry, I've been slammed but I am getting back around to this. I removed raise_for_status from src/api.py since it was catching every HTTP response other than 200, and added the general RequestException test that Sourcery recommended.

As for running everything locally, I followed the instructions to run with Docker but without installing Poetry. When I run the container, it gets a ModuleNotFoundError for flask. I have been trying to look into it but my best guess at this point is that my OS or the virtual environment I am using for the project is causing an issue.

christe6-osu commented:
@sourcery-ai review
